app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java

```diff
@@ -292,6 +327,18 @@ private void initialize() {
                 "Enables old Code Editor from v6.2.0.",
                 SETTING_LEGACY_CODE_EDITOR,
                 false);
+        addSwitchPreference("Automatic open after installing (ROOT)",
+                "\n" +
+                        "Just opens the app automatically after installation",
+                SETTING_ROOTED_AUTOOPEN,
+                true);
+        addTextInputPreference("Grant ROOT access",
+                "Brings app-autoinstallation", v -> {
+                    //TODO
```

Comment:
what's to do here?

Reply:
> what's to do here?

There should be a dialog here in case of an error, for example, if a person simply does not have a ROOT device

Reply:
Okay. I just noticed though, why is it a text input preference?

Reply:
> Okay. I just noticed though, why is it a text input preference?

So, are there other options to open a dialog box with a ROOT request?

Reply:
As far as I know, the app managing root access opens a dialog asking the user if the app should be granted root access. Also, why make it a setting? The user could just allow root access when it's asked for, after building a project and trying to install the APK automatically.

Reply:
I hoped that it would be possible to turn this feature on or off, suddenly it would not work correctly for someone, or simply it would be unusual for a person to use it

Reply:
Then I think the setting should be changed to be a simple setting whether Sketchware Pro should (try to) auto-install APKs with root or not. If it fails after having built the APK, e.g. a Toast could be shown, and the installation be done with Package installer.

Reply:
I understand what you mean, I'll do it later, then I'll redo it in the draft for now

Reply:
That's fine! Thank you for your efforts.